### DISCUSSION TITLE
Bootstrapping the AGW fails

### DISCUSSION THREAD

**Discussion by mcelligottmike** [magma/magma#15155] [Q&A]
Hi
   I have been working through the Quick Start Guide and using magma 1.8.0. I am stuck on the step for Connecting local LTE gateway to Local Cloud. The fab command ran successfully when I ssh into the AGW vagrant image and perform these steps:
HOST [magma/lte/gateway]$ vagrant ssh magma

MAGMA-VM$ sudo service magma@* stop
MAGMA-VM$ sudo service magma@magmad restart
MAGMA-VM$ sudo tail -f /var/log/syslog

I see errors in the syslog included below. It seems the first error is 
Mar 20 16:07:32 magma-dev magmad[1004639]: ERROR:root:GetChallenge error! [StatusCode.UNAVAILABLE] failed to connect to all addresses; last error: UNKNOWN: HTTP proxy returned response code 502

I have looked at the code where this error emanates from but afraid I do not have the expertise to troubleshoot deeper. The one unusual thing about my environment is that I have to deal with a corporate proxy. This required a number of tweaks here and there during installation but thought I was almost there - the webUIs are accessible and visible (NMS one is displaying an error notification due to lack of connectivity between gateway and local cloud as expected) 

Any assistance or advice would be very much appreciated as I am a bit stumped - even to just guess at what might cause the above error

Full error log
==========
Mar 20 16:07:00 magma-dev pipelined[1004970]: INFO:EgressController:Received Setup request with epoch - 1679328414, current epoch  is - 1679328414Mar 20 16:07:00 magma-dev pipelined[1004970]: INFO:EnforcementStatsController:Received Setup request with epoch - 1679328414, current epoch  is - 1679328414
Mar 20 16:07:00 magma-dev pipelined[1004970]: ERROR:OfctlService:unknown barrier xid 1571551837
Mar 20 16:07:00 magma-dev pipelined[1004970]: INFO:root:skipping removal of enforcement flows since it is in stateless mode
Mar 20 16:07:00 magma-dev pipelined[1004970]: INFO:EnforcementController:Initialized proxy_controller <magma.pipelined.app.he.HeaderEnrichmentController object at 0x7ff8521cf8b0>
Mar 20 16:07:00 magma-dev pipelined[1004970]: INFO:GYController:Initialized proxy_controller <magma.pipelined.app.he.HeaderEnrichmentController object at 0x7ff8521cf8b0>
Mar 20 16:07:00 magma-dev pipelined[1004970]: INFO:EnforcementStatsController:Initialized proxy_controller <magma.pipelined.app.he.HeaderEnrichmentController object at 0x7ff8521cf8b0>
Mar 20 16:07:00 magma-dev sessiond[1004739]: I0320 16:07:00.039090 1004739 LocalSessionManagerHandler.cpp:171] Successfully setup PipelineD with epoch: 1679328414
Mar 20 16:07:00 magma-dev pipelined[1004970]: INFO:EgressController:Using GW: mac: aa:d0:6a:7f:f4:4b match {'reg1': 1} actions: [NXActionRegLoad2(dst='eth_dst',experimenter=8992,len=None,mask=None,subtype=33,type=65535,value='aa:d0:6a:7f:f4:4b')]
Mar 20 16:07:00 magma-dev sessiond[1004739]: I0320 16:07:00.042336 1004739 LocalSessionManagerHandler.cpp:171] Successfully setup PipelineD with epoch: 1679328414
Mar 20 16:07:02 magma-dev systemd[1]: Started Magma OAI MME service.
Mar 20 16:07:02 magma-dev sctpd[18115]: I0320 16:07:02.581084 18119 sctpd_downlink_impl.cpp:62] SctpdDownlinkImpl::req->ppid()=18
Mar 20 16:07:02 magma-dev sctpd[18115]: I0320 16:07:02.581097 18119 sctpd_downlink_impl.cpp:64] SctpdDownlinkImpl::req->port()=36412
Mar 20 16:07:02 magma-dev sctpd[18115]: I0320 16:07:02.581100 18119 sctpd_downlink_impl.cpp:40] SctpdDownlinkImpl::Init reusing existing connection
Mar 20 16:07:03 magma-dev smsd[1004837]: INFO:root:mme non_eps_service_config is not SMS_ORC8R, sleeping.
Mar 20 16:07:14 magma-dev smsd[1004837]: message repeated 2 times: [ INFO:root:mme non_eps_service_config is not SMS_ORC8R, sleeping.]
Mar 20 16:07:14 magma-dev magmad[1004639]: INFO:root:Setting logging level to INFO
Mar 20 16:07:19 magma-dev smsd[1004837]: INFO:root:mme non_eps_service_config is not SMS_ORC8R, sleeping.
Mar 20 16:07:29 magma-dev smsd[1004837]: message repeated 2 times: [ INFO:root:mme non_eps_service_config is not SMS_ORC8R, sleeping.]
Mar 20 16:07:32 magma-dev systemd-udevd[1005590]: ethtool: autonegotiation is unset or enabled, the speed and duplex are not writable.
Mar 20 16:07:32 magma-dev systemd-udevd[1005591]: ethtool: autonegotiation is unset or enabled, the speed and duplex are not writable.
Mar 20 16:07:32 magma-dev systemd-udevd[1005591]: Using default interface naming scheme 'v245'.
Mar 20 16:07:32 magma-dev systemd-udevd[1005590]: Using default interface naming scheme 'v245'.
Mar 20 16:07:32 magma-dev systemd-udevd[1005591]: envoy_cntr_ns: Could not generate persistent MAC: No data available
Mar 20 16:07:32 magma-dev systemd-udevd[1005590]: envoy_cntr: Could not generate persistent MAC: No data available
Mar 20 16:07:32 magma-dev magmad[1004639]: ERROR:root:GetChallenge error! [StatusCode.UNAVAILABLE] failed to connect to all addresses; last error: UNKNOWN: HTTP proxy returned response code 502
Mar 20 16:07:32 magma-dev magmad[1004639]: INFO:root:Retrying bootstrap in 30 seconds
Mar 20 16:07:32 magma-dev kernel: [1575235.959009] IPv6: ADDRCONF(NETDEV_CHANGE): envoy_cntr_ns: link becomes ready
Mar 20 16:07:32 magma-dev kernel: [1575235.959039] IPv6: ADDRCONF(NETDEV_CHANGE): envoy_cntr: link becomes ready
Mar 20 16:07:32 magma-dev kernel: [1575235.961248] IPv6: ADDRCONF(NETDEV_CHANGE): proxy_port_ns: link becomes ready
Mar 20 16:07:32 magma-dev envoy[1005627]: net.ipv4.conf.all.rp_filter = 0
Mar 20 16:07:32 magma-dev envoy[1005628]: net.ipv4.conf.all.route_localnet = 1
Mar 20 16:07:32 magma-dev systemd[1]: Started Magma envoy service.
Mar 20 16:07:32 magma-dev envoy_controller[1004770]: I0320 16:07:32.998928 1004770 control_plane.go:133] OnStreamOpen 2 open for
Mar 20 16:07:32 magma-dev envoy_controller[1004770]: I0320 16:07:32.998928 1004770 control_plane.go:133] OnStreamOpen 1 open for type.googleapis.com/envoy.config.cluster.v3.Cluster
Mar 20 16:07:32 magma-dev envoy_controller[1004770]: I0320 16:07:32.999283 1004770 control_plane.go:142] OnStreamRequest
Mar 20 16:07:34 magma-dev smsd[1004837]: INFO:root:mme non_eps_service_config is not SMS_ORC8R, sleeping.
Mar 20 16:07:44 magma-dev smsd[1004837]: message repeated 2 times: [ INFO:root:mme non_eps_service_config is not SMS_ORC8R, sleeping.]
Mar 20 16:07:48 magma-dev envoy_controller[1004770]: I0320 16:07:48.002893 1004770 control_plane.go:133] OnStreamOpen 3 open for type.googleapis.com/envoy.config.listener.v3.Listener
Mar 20 16:07:48 magma-dev envoy_controller[1004770]: I0320 16:07:48.003015 1004770 control_plane.go:142] OnStreamRequest
Mar 20 16:07:49 magma-dev smsd[1004837]: INFO:root:mme non_eps_service_config is not SMS_ORC8R, sleeping.
Mar 20 16:07:52 magma-dev control_proxy[1004759]: 2023-03-20T16:07:52.680Z [127.0.0.1 -> metricsd-controller.magma.test,8443] "POST /magma.orc8r.MetricsController/Collect HTTP/2" 400 237bytes 0.003s
Mar 20 16:07:52 magma-dev control_proxy[1004759]: 2023-03-20T16:07:52.680Z [127.0.0.1 -> metricsd-controller.magma.test,8443] "POST /magma.orc8r.MetricsController/Collect HTTP/2" 400 237bytes 0.003s
Mar 20 16:07:52 magma-dev control_proxy[1004759]: 2023-03-20T16:07:52.680Z [127.0.0.1 -> metricsd-controller.magma.test,8443] "POST /magma.orc8r.MetricsController/Collect HTTP/2" 400 237bytes 0.001s
Mar 20 16:07:52 magma-dev control_proxy[1004759]: message repeated 4 times: [ 2023-03-20T16:07:52.680Z [127.0.0.1 -> metricsd-controller.magma.test,8443] "POST /magma.orc8r.MetricsController/Collect HTTP/2" 400 237bytes 0.001s]Mar 20 16:07:52 magma-dev magmad[1004639]: E0320 16:07:52.685085810 1004662 hpack_parser.cc:1235]       Error parsing metadata: error=invalid value key=content-type value=text/html
Mar 20 16:07:52 magma-dev control_proxy[1004759]: 2023-03-20T16:07:52.680Z [127.0.0.1 -> metricsd-controller.magma.test,8443] "POST /magma.orc8r.MetricsController/Collect HTTP/2" 400 237bytes 0.001s
Mar 20 16:07:52 magma-dev magmad[1004639]: ERROR:root:Metrics upload error for service mobilityd (chunk 0)! [StatusCode.UNKNOWN] Stream removed
Mar 20 16:07:52 magma-dev magmad[1004639]: ERROR:root:Metrics upload error for service subscriberdb (chunk 0)! [StatusCode.UNKNOWN] Stream removed
Mar 20 16:07:52 magma-dev magmad[1004639]: ERROR:root:Metrics upload error for service mme (chunk 0)! [StatusCode.UNKNOWN] Stream removed
Mar 20 16:07:52 magma-dev magmad[1004639]: ERROR:root:Metrics upload error for service magmad (chunk 0)! [StatusCode.UNKNOWN] Stream removed
Mar 20 16:07:52 magma-dev magmad[1004639]: ERROR:root:Metrics upload error for service enodebd (chunk 0)! [StatusCode.UNKNOWN] Stream removed
Mar 20 16:07:52 magma-dev magmad[1004639]: ERROR:root:Metrics upload error for service pipelined (chunk 0)! [StatusCode.UNKNOWN] Stream removed
Mar 20 16:07:52 magma-dev magmad[1004639]: ERROR:root:Metrics upload error for service state (chunk 0)! [StatusCode.UNKNOWN] Stream removed
Mar 20 16:07:52 magma-dev magmad[1004639]: ERROR:root:Metrics upload error for service sessiond (chunk 0)! [StatusCode.UNKNOWN] Stream removed
Mar 20 16:07:53 magma-dev control_proxy[1004759]: 2023-03-20T16:07:53.439Z [127.0.0.1 -> state-controller.magma.test,8443] "POST /magma.orc8r.StateService/ReportStates HTTP/2" 400 237bytes 0.000s
Mar 20 16:07:53 magma-dev magmad[1004639]: ERROR:root:Checkin Error! Failed to report states. [StatusCode.UNKNOWN] Stream removed
Mar 20 16:07:53 magma-dev magmad[1004639]: INFO:root:StateReporting (Checkin) failure threshold met, remediating...
Mar 20 16:07:53 magma-dev magmad[1004639]: INFO:root:Bootstrapping due to invalid certificate
Mar 20 16:07:53 magma-dev magmad[1004639]: ERROR:root:GetChallenge error! [StatusCode.UNAVAILABLE] failed to connect to all addresses; last error: UNKNOWN: HTTP proxy returned response code 502
Mar 20 16:07:53 magma-dev magmad[1004639]: INFO:root:Retrying bootstrap in 30 seconds
Mar 20 16:07:54 magma-dev control_proxy[1004759]: 2023-03-20T16:07:54.018Z [127.0.0.1 -> streamer-controller.magma.test,8443] "POST /magma.orc8r.Streamer/GetUpdates HTTP/2" 400 237bytes 0.001s
Mar 20 16:07:54 magma-dev policydb[1004811]: ERROR:root:Error! Streaming from the cloud failed! [StatusCode.UNKNOWN] Stream removed
Mar 20 16:07:54 magma-dev control_proxy[1004759]: 2023-03-20T16:07:54.018Z [127.0.0.1 -> streamer-controller.magma.test,8443] "POST /magma.orc8r.Streamer/GetUpdates HTTP/2" 400 237bytes 0.001s
Mar 20 16:07:54 magma-dev policydb[1004811]: ERROR:root:Error! Streaming from the cloud failed! [StatusCode.UNKNOWN] Stream removed
Mar 20 16:07:54 magma-dev control_proxy[1004759]: 2023-03-20T16:07:54.018Z [127.0.0.1 -> streamer-controller.magma.test,8443] "POST /magma.orc8r.Streamer/GetUpdates HTTP/2" 400 237bytes 0.001s
Mar 20 16:07:54 magma-dev policydb[1004811]: ERROR:root:Error! Streaming from the cloud failed! [StatusCode.UNKNOWN] Stream removed
Mar 20 16:07:54 magma-dev control_proxy[1004759]: 2023-03-20T16:07:54.018Z [127.0.0.1 -> streamer-controller.magma.test,8443] "POST /magma.orc8r.Streamer/GetUpdates HTTP/2" 400 237bytes 0.001s
Mar 20 16:07:54 magma-dev policydb[1004811]: ERROR:root:Error! Streaming from the cloud failed! [StatusCode.UNKNOWN] Stream removed
Mar 20 16:07:54 magma-dev smsd[1004837]: INFO:root:mme non_eps_service_config is not SMS_ORC8R, sleeping.
Mar 20 16:08:19 magma-dev smsd[1004837]: message repeated 5 times: [ INFO:root:mme non_eps_service_config is not SMS_ORC8R, sleeping.]
Mar 20 16:08:23 magma-dev magmad[1004639]: ERROR:root:GetChallenge error! [StatusCode.UNAVAILABLE] failed to connect to all addresses; last error: UNKNOWN: HTTP proxy returned response code 502
Mar 20 16:08:23 magma-dev magmad[1004639]: INFO:root:Retrying bootstrap in 30 seconds
Mar 20 16:08:24 magma-dev smsd[1004837]: INFO:root:mme non_eps_service_config is not SMS_ORC8R, sleeping.

**Comment by Radula76** [comment]
sudo ovs-vsctl show
if the interfaces have an error, run 'sudo ovs-kmod-upgrade.sh'

**Reply by mcelligottmike** [reply]
Thank you but unfortunately that did not help. I ran the script within the magma VM but it throws an error due to missing python module 'lte'. The VM is now in worse condition so had to destroy and rebuild. Maybe I misinterpreted your instructions?


 sudo ./magma/lte/gateway/deploy/roles/magma/files/ovs-kmod-upgrade.sh
This upgrade would result in datapath level downtime for few minutes!
Do you want to proceed with upgrade ?(y/n)y
Hit:1 http://archive.ubuntu.com/ubuntu focal InRelease
Hit:2 http://security.ubuntu.com/ubuntu focal-security InRelease
Hit:3 http://archive.ubuntu.com/ubuntu focal-updates InRelease
Hit:4 http://archive.ubuntu.com/ubuntu focal-backports InRelease
Get:5 https://download.docker.com/linux/ubuntu xenial InRelease [66.2 kB]
Hit:6 https://linuxfoundation.jfrog.io/artifactory/magma-packages focal-1.8.0 InRelease
Fetched 66.2 kB in 1s (50.7 kB/s)
Reading package lists... Done
Building dependency tree
Reading state information... Done
357 packages can be upgraded. Run 'apt list --upgradable' to see them.
Reading package lists... Done
Building dependency tree
Reading state information... Done
libopenvswitch is already the newest version (2.15.4-9-magma).
openvswitch-common is already the newest version (2.15.4-9-magma).
python3-openvswitch is already the newest version (2.15.4-9-magma).
openvswitch-datapath-dkms is already the newest version (2.15.4-9-magma).
openvswitch-switch is already the newest version (2.15.4-9-magma).
The following package was automatically installed and is no longer required:
  libnetplan0
Use 'sudo apt autoremove' to remove it.
0 upgraded, 0 newly installed, 0 to remove and 357 not upgraded.
ifdown: interface patch-up not configured
 * Detected internal interfaces:
 * Exiting ovsdb-server (257767)
 * Starting ovsdb-server
 * Configuring Open vSwitch system IDs
 * Exiting ovs-vswitchd (257820)
 * Saving interface configuration
 * Removing vport_gtp module
 * Removing openvswitch module
 * Inserting openvswitch module
 * Starting ovs-vswitchd
 * Enabling remote OVSDB managers
 * Restoring interface configuration
 * Finding processes on dead interfaces
RTNETLINK answers: File exists
net.ipv4.ip_forward = 1
ifup: interface patch-up already configured
update successful, installed: 2.15.4-9-magma
Do you want to restart control plane services?
Do you want to proceed with upgrade ?(y/n)y
Traceback (most recent call last):
  File "/usr/local/bin/config_stateless_agw.py", line 25, in <module>
    from lte.protos.mconfig import mconfigs_pb2
ModuleNotFoundError: No module named 'lte'

**Reply by Radula76** [reply]
how are you deploying the AGW? as a VM on kubernetes/cloud or baremetal?

**Reply by mcelligottmike** [reply]
On a VM - I am following the instructions in the Quick Start guide - https://magma.github.io/magma/docs/basics/quick_start_guide

**Reply by Radula76** [reply]
oh, this is the development environment, I haven't started that up myself. 
you may need to look into the control_proxy.yml configuration and networking.  the rootCA.pem certificate, or how the proxy is managing the TLS session.
